source/simulator/initial_conditions.cc

```diff
@@ -303,8 +303,12 @@ namespace aspect
     // overwrite the relevant composition block only
     const unsigned int blockidx = advection_field.block_index(introspection);
     solution.block(blockidx) = particle_solution.block(blockidx);
-    old_solution.block(blockidx) = particle_solution.block(blockidx);
-    old_old_solution.block(blockidx) = particle_solution.block(blockidx);
+
+    if (timestep_number == 0)
```

Comment:
Perhaps write a comment saying that on time step 0 `solution`, `old_solution`, `old_old_solution` are by default the same, but we still need to fill the solution blocks with values.